udtf.go

```diff
@@ -0,0 +1,493 @@
+		ddconn := dconn.(*conn)
+		name := C.CString(name)
+		defer C.free(unsafe.Pointer(name))
+
+		cfunction := pinnedValue[TFT]{
+			pinner: &runtime.Pinner{},
+			value:  function,
+		}
+
+		handle := cgo.NewHandle(cfunction)
+		cfunction.pinner.Pin(&handle)
+		tableFunction := C.duckdb_create_table_function()
+		C.duckdb_table_function_set_name(tableFunction, name)
+		C.duckdb_table_function_set_init(tableFunction, C.init(C.udf_init))
+		C.duckdb_table_function_set_extra_info(tableFunction, unsafe.Pointer(&handle), C.duckdb_delete_callback_t(C.udf_destroy_data))
+		C.duckdb_table_function_supports_projection_pushdown(tableFunction, C.bool(true))
+
+		var config TableFunctionConfig
+		var x any = function
+		switch tfunc := x.(type) {
+		case RowTableFunction:
+			C.duckdb_table_function_set_bind(tableFunction, C.bind(C.udf_bind_row))
+			C.duckdb_table_function_set_function(tableFunction, C.callback(C.udf_row_callback))
+
+			config = tfunc.Config
+			if tfunc.BindArguments == nil {
+				return invalidTableFunctionError()
+			}
+		case ChunkTableFunction:
+			C.duckdb_table_function_set_bind(tableFunction, C.bind(C.udf_bind_chunk))
+			C.duckdb_table_function_set_function(tableFunction, C.callback(C.udf_chunk_callback))
+			config = tfunc.Config
+			if tfunc.BindArguments == nil {
+				return invalidTableFunctionError()
+			}
+		case ThreadedRowTableFunction:
+			C.duckdb_table_function_set_bind(tableFunction, C.bind(C.udf_bind_row))
+			C.duckdb_table_function_set_function(tableFunction, C.callback(C.udf_row_callback))
+			C.duckdb_table_function_set_local_init(tableFunction, C.init(C.udf_local_init))
+			config = tfunc.Config
+			if tfunc.BindArguments == nil {
+				return invalidTableFunctionError()
+			}
+		case ThreadedChunkTableFunction:
+			C.duckdb_table_function_set_bind(tableFunction, C.bind(C.udf_bind_chunk))
+			C.duckdb_table_function_set_function(tableFunction, C.callback(C.udf_chunk_callback))
+			C.duckdb_table_function_set_local_init(tableFunction, C.init(C.udf_local_init))
+			config = tfunc.Config
+			if tfunc.BindArguments == nil {
+				return invalidTableFunctionError()
+			}
+		default:
+			panic("This code should be unreachable, please open a bug report for go-duckdb.")
+		}
+
+		for _, t := range config.Arguments {
+			dt := t.logicalType()
+			defer C.duckdb_destroy_logical_type(&dt)
+			C.duckdb_table_function_add_parameter(tableFunction, dt)
+		}
+
+		for name, t := range config.NamedArguments {
+			dt := t.logicalType()
+			defer C.duckdb_destroy_logical_type(&dt)
+			argName := C.CString(name)
+			defer C.free(unsafe.Pointer(argName))
+			C.duckdb_table_function_add_named_parameter(tableFunction, argName, dt)
+		}
+
+		state := C.duckdb_register_table_function(ddconn.duckdbCon, tableFunction)
+		C.duckdb_destroy_table_function(&tableFunction)
+		if state != 0 {
+			return invalidTableFunctionError()
+		}
+		return nil
+	})
+	return err
+}
+
+func getValue(t TypeInfo, v C.duckdb_value) (any, error) {
```

Comment:
With the recent DuckDB release, getting significantly more values through the C API is now possible.
Should we move this into a separate file (`value.go` or similar) and extend the support for all types?
Otherwise, we can add a `FIXME` and open a follow-up PR.

Reply:
I think this is fine for now, I will add a FIXME